src/js/protocols/stm32.js

```diff
@@ -265,69 +265,71 @@ STM32_protocol.prototype.initialize = function () {
+    const data = new Uint8Array(readInfo.data);
 
-    for (var i = 0; i < data.length; i++) {
-        this.receive_buffer.push(data[i]);
+    for (const instance of data) {
+        this.receive_buffer.push(instance);
     }
 
     // routine that fetches data from buffer if statement is true
-    if (this.receive_buffer.length >= this.bytes_to_read && this.bytes_to_read != 0) {
-        var data = this.receive_buffer.slice(0, this.bytes_to_read); // bytes requested
-        this.receive_buffer.splice(0, this.bytes_to_read); // remove read bytes
+    if (this.receive_buffer.length >= this.bytesToRead && this.bytesToRead != 0) {
+        const fetched = this.receive_buffer.slice(0, this.bytesToRead); // bytes requested
+        this.receive_buffer.splice(0, this.bytesToRead); // remove read bytes
 
-        this.bytes_to_read = 0; // reset trigger
+        this.bytesToRead = 0; // reset trigger
 
-        this.read_callback(data);
+        this.read_callback(fetched);
     }
 };
 
 // we should always try to consume all "proper" available data while using retrieve
 STM32_protocol.prototype.retrieve = function (n_bytes, callback) {
     if (this.receive_buffer.length >= n_bytes) {
         // data that we need are there, process immediately
-        var data = this.receive_buffer.slice(0, n_bytes);
+        const data = this.receive_buffer.slice(0, n_bytes);
         this.receive_buffer.splice(0, n_bytes); // remove read bytes
 
         callback(data);
     } else {
         // still waiting for data, add callback
-        this.bytes_to_read = n_bytes;
+        this.bytesToRead = n_bytes;
         this.read_callback = callback;
     }
 };
 
 // bytes_to_send = array of bytes that will be send over serial
-// bytes_to_read = received bytes necessary to trigger read_callback
-// callback = function that will be executed after received bytes = bytes_to_read
-STM32_protocol.prototype.send = function (bytes_to_send, bytes_to_read, callback) {
+// bytesToRead = received bytes necessary to trigger read_callback
+// callback = function that will be executed after received bytes = bytesToRead
+STM32_protocol.prototype.send = function (bytes_to_send, bytesToRead, callback) {
     // flip flag
     this.upload_process_alive = true;
 
-    var bufferOut = new ArrayBuffer(bytes_to_send.length);
-    var bufferView = new Uint8Array(bufferOut);
+    const bufferOut = new ArrayBuffer(bytes_to_send.length);
+    const bufferView = new Uint8Array(bufferOut);
 
     // set bytes_to_send values inside bufferView (alternative to for loop)
     bufferView.set(bytes_to_send);
 
     // update references
-    this.bytes_to_read = bytes_to_read;
+    this.bytesToRead = bytesToRead;
     this.read_callback = callback;
 
     // empty receive buffer before next command is out
     this.receive_buffer = [];
 
     // send over the actual data
-    serial.send(bufferOut, function (writeInfo) {});
+    serial.send(bufferOut, function (writeInfo) {
+        //
```

Comment:
Looking at `src/js/serial.js`, `callback` seems to be optional, so we can probably just omit this.

Reply:
Reduced.